Src/java/quick/src/main/resources/org/hl7/fhir/fhir-modelinfo-4.0.0.xml

```diff
@@ -299,7 +299,7 @@
         <ns4:element name="availableStartTime" elementType="FHIR.time"/>
         <ns4:element name="availableEndTime" elementType="FHIR.time"/>
     </ns4:typeInfo>
-    <ns4:typeInfo xsi:type="ns4:ClassInfo" name="DeviceUseStatement" retrievable="true" baseType="FHIR.DomainResource">
+    <ns4:typeInfo xsi:type="ns4:ClassInfo" name="DeviceUseStatement" retrievable="true" baseType="FHIR.DomainResource" primaryCodePath="device.code">
```

Comment:
So the `primaryCodePath` is intended to support automatic resolution of FHIR references?  This feels a little bit like magic, because as far as the model-info is concerned, `device` is a `Reference`, and `Reference` is just another class -- but `code` is not a valid property of that class.  Resolving this requires special undocumented behavior -- and also opens the door for ambiguity (e.g., how would you reference an actual *property* of `Reference` as opposed to the thing it refers to?).

If we need to support something like this, I think it may be worth making the jump and saying that `primaryCodePath` is actually a FHIRPath expression.  Then things like reference resolution are intentionally declared (e.g., `resolve()`) and the semantics/behavior of the code path is formally specified.

(FHIR 4.0.0 `DeviceUseStatement`)

Reply:
Agreed that the behavior should be explicit. There is a restricted subset of FHIRPath that is used in the FHIR spec for this exact purpose, we should restrict it to that same set: http://build.fhir.org/fhirpath.html#simple

Until we have that though, I suggest we leave the primaryCodePath unspecified then.